src/backend/distributed/operations/repair_shards.c

```diff
@@ -1188,9 +1188,26 @@ CopyShardTablesViaBlockWrites(List *shardIntervalList, char *sourceNodeName,
 
 	/*
 	 * Once all shards are created, we can recreate relationships between shards.
-	 *
-	 * Iterate through the colocated shards and create the foreign constraints and
-	 * attach child tables to their parents in a partitioning hierarchy.
+	 * Attach child tables to their parents in a partitioning hierarchy.
+	 */
+	foreach_ptr(shardInterval, shardIntervalList)
+	{
+		if (PartitionTable(shardInterval->relationId))
+		{
+			char *attachPartitionCommand =
+				GenerateAttachShardPartitionCommand(shardInterval);
+
+			char *tableOwner = TableOwner(shardInterval->relationId);
+			SendCommandListToWorkerOutsideTransaction(targetNodeName, targetNodePort,
```

Comment:
I kind of wonder whether we should try to do fewer transactions by building 1 big list to reduce the number of possible outcomes after a failure, but not really this PR's fault.

Reply:
Will refactor this. Not too much of work.

Reply:
Upated but there seems to be a failure around shard size. Need to investigate further. https://app.circleci.com/pipelines/github/citusdata/citus/25160/workflows/be050c43-e08b-4bb5-9550-d48359c1a48d/jobs/695686/steps

Reply:
The new output seems to make more sense, but a bit curious that it's different.

Having the raw sizes in there might be a sketchy in general, since PG gives few guarantees about exact size.